atlasdb-autobatch/src/main/java/com/palantir/atlasdb/autobatch/DisruptorAutobatcher.java

```diff
@@ -36,10 +38,19 @@
  */
 public final class DisruptorAutobatcher<T, R>
         implements AsyncFunction<T, R>, Function<T, ListenableFuture<R>>, Closeable {
-    private static final int BUFFER_SIZE = 1024;
+
+    /*
+        By memoizing thread factories per loggable purpose, the names are correct for multiple instances of the same
+        autobatcher function.
+     */
+    private static final ConcurrentMap<String, ThreadFactory> threadFactories = Maps.newConcurrentMap();
 
     private static ThreadFactory threadFactory(String safeLoggablePurpose) {
-        String namePrefix = String.format("autobatcher-%s-", safeLoggablePurpose);
+        return threadFactories.computeIfAbsent(safeLoggablePurpose, DisruptorAutobatcher::createThreadFactory);
+    }
+
+    private static ThreadFactory createThreadFactory(String safeLoggablePurpose) {
+        String namePrefix = String.format("autobatcher.%s-", safeLoggablePurpose);
```

Comment:
nit: I think this change from `-` to `.` is intentional, given what you're doing in `AutobatchingPaxosPingableLeaderFactory`. This seems good to allow clear delimiting between multi-word phrases. In that case, `BatchingPaxosAcceptorFactory` should probably be updated to use the new format as well (as in `batch-paxos-acceptor.latest-sequence-cache`, for instance)

Reply:
done, eagle eyed! 🦅 👁